### DISCUSSION TITLE
Flash memory footprint

### DISCUSSION THREAD

**Discussion by AIM65** [schreibfaul1/ESP32-audioI2S#712] [Q&A]
This is a great software, I'm currently building a webradio around it.
Is there a simple, and clean, way to reduce program footprint by disabling some features ? I Mainly need MP3 and AAC, nothing more and my ESP has 4Meg of flash and no PSRAM. Easier way is to use the right ESP such a C or WROVER with PSRAM and 8Meg, but my WROOM is already installed on a custom pcb... !
By 'clean' I mean doing some changes in the source code at specific locations in order to be able to safely reproduce them into the future releases but no large rework.

And, by the way, if changing the ESP is the best solution, as it's not that difficult and expensive to update the PCB, which is currently the right ESP to choose in order to correctly handle all the features of the software ?

**Comment by whitevast** [comment]
Yes, it would be nice to make codecs switchable. This will save memory space and make it possible to use OTA, which would allow you to update the firmware without opening the device.

**Comment by stef-ladefense** [comment]
look there : https://github.com/stef-ladefense/ESP32-audioI2S
you can choise codec in audio.h

**Reply by whitevast** [reply]
Oh! That's what you need. Thank you very much!
